tests/base/src/test/java/org/wildfly/security/sasl/test/SaslServerBuilder.java

```diff
@@ -298,6 +307,9 @@ public SaslServer build() throws IOException {
             securityDomainReference.setReference(securityDomain);
         }
         SaslServerFactory factory = obtainSaslServerFactory(serverFactoryClass);
+        if (providerSupplier != null) {
```

Comment:
I think it would make sense to change this check to `if (factory == null && providerSupplier != null)` so that we'll attempt to use the `SecurityProviderSaslServerFactory` if we weren't already able to obtain the SASL server factory we wanted.